### PR TITLE
Add a small tutorial on deploying Quarkus app with Ansible

### PR DESCRIPTION
**Pull request by rpelisse**

### CODE REVIEW

**Comment by rpelisse** [comment]
@michalvavrik the preview looks good... Any comments on your side?

File: docs/src/main/asciidoc/ansible.adoc [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+
+Here again, it checks that the appropriate OpenJDK is available on the target system. Then we ensure that the required user and group exist on the target. This is recommended mostly to be able to run the Quarkus app with a regular user, rather than with the root account.
+
+With those requirements in place, the jar can be deployed on the target, along with the required configuration for the app integration into systemd as a service. Any change to the systemd configuration requires reloading its daemon, which the collection ensures will happen whenever it is needed. With all of that in place, the only thing that remains is to start the service itself, which Ansible will also take care of.
+
+=== Validate that deployment was successful
```
Comment (michalvavrik) [inline docs/src/main/asciidoc/ansible.adoc:139]: Automate deployment and validate success manually seems really miserable, I suppose if this is ever going to be used seriously, users will ping readyness/liveness endpoints (probes)? Doesn't Ansible provide verification/validation that could ping these endpoints in last step?

**Review by michalvavrik — COMMENTED** [review]
Thanks @rpelisse , I've left few comments. I also think 2 minutes for reading is very optimistic, but let say I'm just slow reader.

File: docs/src/main/asciidoc/ansible.adoc [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+		-e quarkus_app_repo_url='https://github.com/quarkusio/quarkus-quickstarts.git' \
+		-e quarkus_app_source_folder='getting-started' \
+		-e quarkus_path_to_folder_to_deploy=/opt/quarkus_deploy
+----
+
+The four parameters provided to the playbook are pretty self-explaratory. The first one, `app_name`, is the name of the application, in our case, it's just `getting-started`. The second one, `quarkus_app_repo_url`, is the URL to the Git repository to checkout. The third one is optional, `quarkus_app_source_folder` speficifies, if needed, which subfolder from the repo the source code is located. Finally, the fourth one, simply indicates where to install the application on the target.
+
+=== Inventory file
+
+If you are not familiar with Ansible, please note that que file inventory needs to be provided for the command above to run properly. This is a simple text file providing the information Ansible requires on the target system it manages. Please refer to the Ansible documentation for more information on https://docs.ansible.com/ansible/latest/inventory_guide/intro_inventory.html[Ansible inventory].
```
[the remark this answers is not in the document]
Comment (rpelisse) [reply docs/src/main/asciidoc/ansible.adoc:44]: que is an alternative spelling for the, when my keyboard screws with me :D

**Comment by michalvavrik** [comment]
thanks @rpelisse, let's also squash the commits

File: docs/src/main/asciidoc/ansible.adoc (hunk 1 above)
Comment (rpelisse) [reply docs/src/main/asciidoc/ansible.adoc:139]: It's a tutorial, I'm giving the reader the tools to double check that what the stuff actually worked, it's not a deployment design. I'll rephrase that.

Pinging the endpoint is trivial with Ansible, you have a primitive for that (ansible.builtin.get_url), but I can add that too.

Comment (michalvavrik) [reply docs/src/main/asciidoc/ansible.adoc:139]: I thought this paragraph was very far from real world, but you will know better how users are going to use this. Would it be too much of a trouble to add ping to the playbook and mention it in the text? But sure, please treat this as opinionated preference, nothing else!

**Comment by michalvavrik** [comment]
@geoand please could you assign a suitable reviewer? I don't know who has closes to the topic, thank you

**Comment by rpelisse** [comment]
@michalvavrik I've integrated your suggestions, bump the tutorial duration to 10', added the part about automating the validation. I just need to write up the part of the playbook associated to that (hence the TODO, there). 

Let me know if I missed any other feedback from you!

**Comment by michalvavrik** [comment]
Thanks for changes @rpelisse , when you resolve TODO and squash commits to one, the PR will be good in my eyes. We will need additional reviewer, Georgios will find one.

File: docs/src/main/asciidoc/ansible.adoc [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+[source,bash]
+----
+$ ansible-playbook -i inventory playbook.yml
+----
+
+You also can automate the validation part we mentioned in the previous section. You can use the https://docs.ansible.com/ansible/latest/collections/ansible/builtin/assert_module.html[ansible.builtin.assert] module and populate the https://docs.ansible.com/ansible/latest/collections/ansible/builtin/service_facts_module.html#examples[service facts] to ensure the systemd service of the Quarkus app is running, along with https://docs.ansible.com/ansible/latest/collections/ansible/builtin/assert_module.html[ansible.builtin.get_url] to check that the webapp is accessible.
+
+```
+ post_tasks:
+   # TODO
```
[the remark this answers is not in the document]
Comment (rpelisse) [reply docs/src/main/asciidoc/ansible.adoc:221]: @machi1990 see [comment above](https://github.com/quarkusio/quarkus/pull/32851#issuecomment-1519882886).

File: docs/src/main/asciidoc/ansible.adoc [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+----
+
+To run this playbook, you use again the ansible-playbook command, but providing now the path to the playbook:
+
+[source,bash]
+----
+$ ansible-playbook -i inventory playbook.yml
+----
+
+You also can automate the validation part we mentioned in the previous section. You can use the https://docs.ansible.com/ansible/latest/collections/ansible/builtin/assert_module.html[ansible.builtin.assert] module and populate the https://docs.ansible.com/ansible/latest/collections/ansible/builtin/service_facts_module.html#examples[service facts] to ensure the systemd service of the Quarkus app is running, along with https://docs.ansible.com/ansible/latest/collections/ansible/builtin/assert_module.html[ansible.builtin.get_url] to check that the webapp is accessible.
```
[the remark this answers is not in the document]
Comment (rpelisse) [reply docs/src/main/asciidoc/ansible.adoc:217]: I put ReST app into web app, but if you feel it's wrong, I can change that.

File: docs/src/main/asciidoc/ansible.adoc (hunk 3 above)
[the remark this answers is not in the document]
Comment (machi1990) [reply docs/src/main/asciidoc/ansible.adoc:221]: Thanks, I missed that.

[the remark this answers is not in the document]
Comment (rpelisse) [reply docs/src/main/asciidoc/ansible.adoc:221]: No issue! I'm finishing the playbook as we speak, I just wanted you guys to be able to review the rest in the meantime.

File: docs/src/main/asciidoc/ansible.adoc (hunk 4 above)
[the remark this answers is not in the document]
Comment (rpelisse) [reply docs/src/main/asciidoc/ansible.adoc:217]: I've replaced webapp by Quarkus app.

**Comment by geoand** [comment]
@jacobdotcosta perhaps you can review this one?

File: docs/src/main/asciidoc/ansible.adoc (hunk 4 above)
[the remark this answers is not in the document]
Comment (michalvavrik) [reply docs/src/main/asciidoc/ansible.adoc:217]: thx, even better

**Comment by rpelisse** [comment]
@geoand @jacobdotcosta FYI, I've added the missing playbook. Unless I've overlooked something or you have new feedback, this PR is ready.

**Comment by jacobdotcosta** [comment]
I'll take a look.

**Comment by rpelisse** [comment]
@jacobdotcosta  Rebasing and squashing commits appears to have confused the CI tooling, so I've setup a new [PR](https://github.com/quarkusio/quarkus/pull/32860).

**Pull request closed without merging**